config.go

```diff
@@ -92,6 +98,7 @@ type Config struct {
 	// used to Initialize, Validate
 	configHandlers []configHandler
 
+	Dialer      Dialer        `opt:"dialer"`
 	DialTimeout time.Duration `opt:"dial_timeout" default:"1s"`
```

Comment:
maybe these should be moved next to `LocalAddr` (just below) and get a little doc comment?